SamsungPay/src/main/java/com/braintreepayments/api/SamsungPayInternalClient.java

```diff
@@ -0,0 +1,164 @@
+                .setConfiguration(configuration)
+                .setSessionId(braintreeClient.getSessionId())
+                .setIntegrationType(braintreeClient.getIntegrationType())
+                .build());
+    }
+
+    private SamsungPayInternalClient(Context context, Configuration configuration, PartnerInfo partnerInfo) {
+        this(configuration, new SamsungPay(context, partnerInfo), new PaymentManager(context, partnerInfo));
+    }
+
+    @VisibleForTesting
+    SamsungPayInternalClient(Configuration configuration, SamsungPay samsungPay, PaymentManager paymentManager) {
+        this.braintreeAcceptedCardBrands = mapToSamsungPayCardBrands(configuration.getSupportedCardTypes());
+        this.samsungPay = samsungPay;
+        this.paymentManager = paymentManager;
+    }
+
+    void goToSamsungPayUpdatePage() {
+        samsungPay.goToUpdatePage();
+    }
+
+    void activateSamsungPay() {
+        samsungPay.activateSamsungPay();
+    }
+
+    void updateCustomSheet(CustomSheet customSheet) {
+        paymentManager.updateSheet(customSheet);
+    }
+
+    void getSamsungPayStatus(final GetSamsungPayStatusCallback callback) {
+        samsungPay.getSamsungPayStatus(new StatusListener() {
+            @Override
+            public void onSuccess(int statusCode, Bundle bundle) {
+                Exception samsungPayError = null;
+
+                @SamsungPayError int error = SamsungPayError.SAMSUNG_PAY_ERROR_UNKNOWN;
+                if (statusCode != SPAY_READY) {
+                    if (bundle != null && bundle.containsKey(SamsungPay.EXTRA_ERROR_REASON)) {
+                        int reason = bundle.getInt(SamsungPay.EXTRA_ERROR_REASON);
+                        switch (reason) {
+                            case ERROR_SPAY_APP_NEED_TO_UPDATE:
+                                error = SamsungPayError.SAMSUNG_PAY_APP_NEEDS_UPDATE;
+                                break;
+                            case ERROR_SPAY_SETUP_NOT_COMPLETED:
+                                error = SamsungPayError.SAMSUNG_PAY_SETUP_NOT_COMPLETED;
+                                break;
+                        }
+                    } else {
+                        if (statusCode == SPAY_NOT_READY) {
+                            error = SamsungPayError.SAMSUNG_PAY_NOT_READY;
+                        } else {
+                            // SPAY_NOT_SUPPORTED
+                            error = SamsungPayError.SAMSUNG_PAY_NOT_SUPPORTED;
+                        }
+                    }
+                    samsungPayError = new SamsungPayException(error);
+                }
+
+                callback.onResult(statusCode, samsungPayError);
+            }
+
+            @Override
+            public void onFail(int errorCode, Bundle bundle) {
+                SamsungPayException exception = new SamsungPayException(errorCode);
+                callback.onResult(null, exception);
```

Comment:
Why not call back the error code instead of null here?

Reply:
Yeah the assumption I went with here was a `null` value communicates via callback that an Error occurred. Also not sure if there could be a potential collision between SamsungPay status codes and SamsungPay error codes.